packages/atomic/src/components/search/atomic-result-timespan/atomic-result-timespan.ts

```diff
@@ -0,0 +1,129 @@
+import {Schema, StringValue} from '@coveo/bueno';
+import {type Result, ResultTemplatesHelpers} from '@coveo/headless';
+import dayjs from 'dayjs';
+import type {DurationUnitType} from 'dayjs/plugin/duration';
+import duration from 'dayjs/plugin/duration';
+import {html, LitElement} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
+import {ValidatePropsController} from '@/src/components/common/validate-props-controller/validate-props-controller';
+import type {Bindings} from '@/src/components/search/atomic-search-interface/interfaces';
+import {createResultContextController} from '@/src/components/search/result-template-component-utils/context/result-context-controller';
+import {bindingGuard} from '@/src/decorators/binding-guard';
+import {bindings} from '@/src/decorators/bindings';
+import {errorGuard} from '@/src/decorators/error-guard';
+import type {InitializableComponent} from '@/src/decorators/types';
+import {InitializeBindingsMixin} from '@/src/mixins/bindings-mixin';
+import {LightDomMixin} from '@/src/mixins/light-dom';
+
+dayjs.extend(duration);
+
+/**
+ * The `atomic-result-timespan` component renders a target result number field value as a duration.
+ */
+@customElement('atomic-result-timespan')
+@bindings()
+export class AtomicResultTimespan
+  extends LightDomMixin(InitializeBindingsMixin(LitElement))
+  implements InitializableComponent<Bindings>
+{
+  /**
+   * The target result field.
+   * The component first looks for the field in the Result object, and then in the Result.raw object.
+   * It is important to include the necessary field in the `atomic-search-interface` component.
+   */
+  @property({reflect: true}) public field!: string;
+  /**
+   * The unit of measurement of the field value.
+   * Available units: https://day.js.org/docs/en/durations/creating
+   */
+  @property({reflect: true}) public unit = 'ms';
+  /**
+   * The format to apply to the result field value.
+   *
+   * By default, the format is HH:mm:ss when the duration is under a day, and it is an approximation when longer (days, months or years).
+   *
+   * The string displayed when there is an approximation can be modified with [localization](https://docs.coveo.com/en/atomic/latest/usage/atomic-localization/).
+   *
+   * Available formats: https://day.js.org/docs/en/durations/format
+   */
+  @property({reflect: true}) public format?: string;
+
+  @state() public bindings!: Bindings;
+  @state() public error!: Error;
+
+  private resultContext = createResultContextController(this);
+
+  constructor() {
+    super();
+
+    new ValidatePropsController(
+      this,
+      () => ({field: this.field}),
+      new Schema({
+        field: new StringValue({required: true, emptyAllowed: false}),
+      })
+    );
+  }
+
+  public initialize() {
+    if (!this.value) {
+      this.error = new Error(`No value found for field ${this.field}`);
+      return;
+    }
+    if (Number.isNaN(this.value)) {
```

Comment:
There's a potential subtle regression here.

`Number.isNaN('abc')` will return `false`, while `isNaN('abc')` (which was used in the Stencil version of the component) will return `true` (you can test it in a JS console).

We should use `isNaN(this.value)` instead to preserve the behavior. We should also cover the case where we pass something like 'abc' as a value in the unit tests to ensure we don't accidentally introduce the migration later.